src/main/kotlin/lotto/domain/Prize.kt

```diff
@@ -1,24 +1,16 @@
 package lotto.domain
 
-enum class Prize(val countOfMatch: Int, private val rank: Int, val prizeMoney: Int, var count: Int = 0) {
-    MATCH_NONE(0, 0, 0),
-    MATCH_ONE(1, 0, 0),
-    MATCH_TWO(2, 0, 0),
+enum class Prize(val countOfMatch: Int, private val rank: Int, val prizeMoney: Int, val withBonus: Boolean = false) {
     MATCH_THREE(3, 5, 5_000),
     MATCH_FOUR(4, 4, 50_000),
     MATCH_FIVE(5, 3, 1_500_000),
-    MATCH_ALL(6, 1, 2_000_000_000);
-
-    fun addCount() {
-        count++
-    }
+    MATCH_FIVE_WITH_BONUS(5, 2, 30_000_000, true),
+    MATCH_ALL(6, 1, 2_000_000_000),
+    MISS(0, 0, 0);
 
     companion object {
-        fun getPrizeMoney(countOfMatch: Int): Int {
-            return values().find { it.countOfMatch == countOfMatch }?.prizeMoney ?: 0
-        }
-
         fun getPrize(countOfMatch: Int): Prize {
+            if (countOfMatch < MATCH_THREE.countOfMatch) return MISS
             return values().find { it.countOfMatch == countOfMatch }!!
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
            return values().find { it.countOfMatch == countOfMatch } ?: MISS
```